# Charte graphique

## <p style="background-color:#1A5276; color:#FFFFFF; padding:15px; border-radius:6px; font-size:22px; font-weight:bold;">1. Section principale</p>

### <p style="background-color:#2980B9; color:#FFFFFF; padding:10px 15px; border-radius:5px; font-size:18px; font-weight:bold;">1.1 Sous-section</p>

#### <p style="background-color:#AED6F1; color:#1A5276; padding:8px 15px; border-radius:4px; font-size:15px; font-weight:bold;">1.1.1 Sous-sous-section</p>

##### <p style="background-color:#b8d1e2;color:black; padding:8px 15px; border-radius:4px; font-size:15px; font-weight:bold;">1.1.1.x.x.x. Toutes les autres sous-section</p>

# <p style="background: linear-gradient(90deg, #1A5276, #2980B9); color:#FFFFFF; padding:30px; border-radius:10px; text-align:center;"><span style="font-size:34px; font-weight:bold;">Identifiez les causes d'attrition au sein de l'ESN TechNova Partners</span><br></p>

## <p style="background-color:#1A5276; color:#FFFFFF; padding:15px; border-radius:6px; font-size:22px; font-weight:bold;">0. Import</p>

### <p style="background-color:#2980B9; color:#FFFFFF; padding:10px 15px; border-radius:5px; font-size:18px; font-weight:bold;">0.1 Import librairie</p>

In [ ]:
import pandas as pd
import plotly.express as px
import numpy as np
import plotly.graph_objects as go

### <p style="background-color:#2980B9; color:#FFFFFF; padding:10px 15px; border-radius:5px; font-size:18px; font-weight:bold;">0.2 - Import des DataSet</p>

In [34]:
eval = pd.read_csv("data/extrait_eval.csv")
sirh = pd.read_csv("data/extrait_sirh.csv")
sondage = pd.read_csv("data/extrait_sondage.csv")

## <p style="background-color:#1A5276; color:#FFFFFF; padding:15px; border-radius:6px; font-size:22px; font-weight:bold;">1. - DataSet extrait_eval.csv</p>

### <p style="background-color:#2980B9; color:#FFFFFF; padding:10px 15px; border-radius:5px; font-size:18px; font-weight:bold;">1.1 - Analyse préliminaire</p>

In [35]:
eval.info()

<class 'pandas.DataFrame'>
RangeIndex: 1470 entries, 0 to 1469
Data columns (total 10 columns):
 #   Column                                     Non-Null Count  Dtype
---  ------                                     --------------  -----
 0   satisfaction_employee_environnement        1470 non-null   int64
 1   note_evaluation_precedente                 1470 non-null   int64
 2   niveau_hierarchique_poste                  1470 non-null   int64
 3   satisfaction_employee_nature_travail       1470 non-null   int64
 4   satisfaction_employee_equipe               1470 non-null   int64
 5   satisfaction_employee_equilibre_pro_perso  1470 non-null   int64
 6   eval_number                                1470 non-null   str  
 7   note_evaluation_actuelle                   1470 non-null   int64
 8   heure_supplementaires                      1470 non-null   str  
 9   augementation_salaire_precedente           1470 non-null   str  
dtypes: int64(7), str(3)
memory usage: 115.0 KB


Toutes les valeur sont remplies aucune vide

In [36]:
eval.describe().T

,count,mean,std,min,25%,50%,75%,max
satisfaction_employee_environnement,1470.0,2.721769,1.093082,1.0,2.0,3.0,4.0,4.0
note_evaluation_precedente,1470.0,2.729932,0.711561,1.0,2.0,3.0,3.0,4.0
niveau_hierarchique_poste,1470.0,2.063946,1.106940,1.0,1.0,2.0,3.0,5.0
satisfaction_employee_nature_travail,1470.0,2.728571,1.102846,1.0,2.0,3.0,4.0,4.0
satisfaction_employee_equipe,1470.0,2.712245,1.081209,1.0,2.0,3.0,4.0,4.0
satisfaction_employee_equilibre_pro_perso,1470.0,2.761224,0.706476,1.0,2.0,3.0,3.0,4.0
note_evaluation_actuelle,1470.0,3.153741,0.360824,3.0,3.0,3.0,3.0,4.0


In [37]:
eval.head(2)

,satisfaction_employee_environnement,note_evaluation_precedente,niveau_hierarchique_poste,satisfaction_employee_nature_travail,satisfaction_employee_equipe,satisfaction_employee_equilibre_pro_perso,eval_number,note_evaluation_actuelle,heure_supplementaires,augementation_salaire_precedente
0,2,3,2,4,1,1,E_1,3,Oui,11 %
1,3,2,2,2,4,3,E_2,4,Non,23 %


In [38]:
for col in eval.columns:
    print(f" {col} : {eval[col].unique()}\n============================")

 satisfaction_employee_environnement : [2 3 4 1]
 note_evaluation_precedente : [3 2 4 1]
 niveau_hierarchique_poste : [2 1 3 4 5]
 satisfaction_employee_nature_travail : [4 2 3 1]
 satisfaction_employee_equipe : [1 4 2 3]
 satisfaction_employee_equilibre_pro_perso : [1 3 2 4]
 eval_number : <StringArray>
[   'E_1',    'E_2',    'E_4',    'E_5',    'E_7',    'E_8',   'E_10',
   'E_11',   'E_12',   'E_13',
 ...
 'E_2054', 'E_2055', 'E_2056', 'E_2057', 'E_2060', 'E_2061', 'E_2062',
 'E_2064', 'E_2065', 'E_2068']
Length: 1470, dtype: str
 note_evaluation_actuelle : [3 4]
 heure_supplementaires : <StringArray>
['Oui', 'Non']
Length: 2, dtype: str
 augementation_salaire_precedente : <StringArray>
['11 %', '23 %', '15 %', '12 %', '13 %', '20 %', '22 %', '21 %', '17 %',
 '14 %', '16 %', '18 %', '19 %', '24 %', '25 %']
Length: 15, dtype: str


##### <p style="background-color:#b8d1e2;color:black; padding:8px 15px; border-radius:4px; font-size:15px; font-weight:bold;">1.2.1 - Tableaux des variables</p>

| Variable| Descritpion | A garder ? |valeur type |
|:-------|:------|:-------:|-------:  |
| satisfaction_employee_environnement|Note de satisfaction sur l'environement|oui|1 à 4|
| note_evaluation_precedente     |Peut être la note donné par le responssable|  non |1 à 4  |
| niveau_hierarchique_poste     |Peut être le niveau de hierachie du poste| oui  | 1 à 5 |
| satisfaction_employee_nature_travail     |Note donné pour jugé si ce qu'il fait lui plait| oui  |1 à 4|
| satisfaction_employee_equipe      |Note pour la relation equipe| oui|1 à 4  |
| satisfaction_employee_equilibre_pro_perso      |Note pour l'mpietement du pro sur le perso| oui |1 à 4|
| eval_number      |Numéro de l'évaliation | non |E_1 / E_2  |
| note_evaluation_actuelle     |idem note precedente  |non| 3 ou 4|
| heure_supplementaires     | S'il fait des heures sup  |oui       |oui / non |
| augementation_salaire_precedente     |Valeur de l'augementation|oui  |en %     |

In [39]:
eval_clean = eval.drop(columns=["note_evaluation_precedente","eval_number","note_evaluation_actuelle"])

### <p style="background-color:#2980B9; color:#FFFFFF; padding:10px 15px; border-radius:5px; font-size:18px; font-weight:bold;">1.2 - Répartition des notes</p>

In [40]:
# apply value_counts() sur chaque colonne séparément
df_counts = eval[['satisfaction_employee_environnement','satisfaction_employee_nature_travail',
       'satisfaction_employee_equipe',
       'satisfaction_employee_equilibre_pro_perso']].apply(pd.Series.value_counts)
df_counts

,satisfaction_employee_environnement,satisfaction_employee_nature_travail,satisfaction_employee_equipe,satisfaction_employee_equilibre_pro_perso
1,284,289,276,80
2,287,280,303,344
3,453,442,459,893
4,446,459,432,153


In [41]:
fig = px.bar(df_counts, barmode="group")
fig.show()

## <p style="background-color:#1A5276; color:#FFFFFF; padding:15px; border-radius:6px; font-size:22px; font-weight:bold;">2. - DataSet extrait_sirh.csv</p>

### <p style="background-color:#2980B9; color:#FFFFFF; padding:10px 15px; border-radius:5px; font-size:18px; font-weight:bold;">2.1 - Analyse préliminaire</p>

In [82]:
sirh.info()

<class 'pandas.DataFrame'>
RangeIndex: 1470 entries, 0 to 1469
Data columns (total 12 columns):
 #   Column                          Non-Null Count  Dtype
---  ------                          --------------  -----
 0   id_employee                     1470 non-null   int64
 1   age                             1470 non-null   int64
 2   genre                           1470 non-null   str  
 3   revenu_mensuel                  1470 non-null   int64
 4   statut_marital                  1470 non-null   str  
 5   departement                     1470 non-null   str  
 6   poste                           1470 non-null   str  
 7   nombre_experiences_precedentes  1470 non-null   int64
 8   nombre_heures_travailless       1470 non-null   int64
 9   annee_experience_totale         1470 non-null   int64
 10  annees_dans_l_entreprise        1470 non-null   int64
 11  annees_dans_le_poste_actuel     1470 non-null   int64
dtypes: int64(8), str(4)
memory usage: 137.9 KB


In [83]:
sirh.describe().T

,count,mean,std,min,25%,50%,75%,max
id_employee,1470.0,1024.865306,602.024335,1.0,491.25,1020.5,1555.75,2068.0
age,1470.0,36.923810,9.135373,18.0,30.00,36.0,43.00,60.0
revenu_mensuel,1470.0,6502.931293,4707.956783,1009.0,2911.00,4919.0,8379.00,19999.0
nombre_experiences_precedentes,1470.0,2.693197,2.498009,0.0,1.00,2.0,4.00,9.0
nombre_heures_travailless,1470.0,80.000000,0.000000,80.0,80.00,80.0,80.00,80.0
annee_experience_totale,1470.0,11.279592,7.780782,0.0,6.00,10.0,15.00,40.0
annees_dans_l_entreprise,1470.0,7.008163,6.126525,0.0,3.00,5.0,9.00,40.0
annees_dans_le_poste_actuel,1470.0,4.229252,3.623137,0.0,2.00,3.0,7.00,18.0


In [84]:
sirh.head(2)

,id_employee,age,genre,revenu_mensuel,statut_marital,departement,poste,nombre_experiences_precedentes,nombre_heures_travailless,annee_experience_totale,annees_dans_l_entreprise,annees_dans_le_poste_actuel
0,1,41,F,5993,Célibataire,Commercial,Cadre Commercial,8,80,8,6,4
1,2,49,M,5130,Marié(e),Consulting,Assistant de Direction,1,80,10,10,7


In [85]:
for col in sirh.columns:
    print(f" {col} : {sirh[col].unique()}\n============================")

 id_employee : [   1    2    4 ... 2064 2065 2068]
 age : [41 49 37 33 27 32 59 30 38 36 35 29 31 34 28 22 53 24 21 42 44 46 39 43
 50 26 48 55 45 56 23 51 40 54 58 20 25 19 57 52 47 18 60]
 genre : <StringArray>
['F', 'M']
Length: 2, dtype: str
 revenu_mensuel : [5993 5130 2090 ... 9991 5390 4404]
 statut_marital : <StringArray>
['Célibataire', 'Marié(e)', 'Divorcé(e)']
Length: 3, dtype: str
 departement : <StringArray>
['Commercial', 'Consulting', 'Ressources Humaines']
Length: 3, dtype: str
 poste : <StringArray>
[       'Cadre Commercial',  'Assistant de Direction',
              'Consultant',               'Tech Lead',
                 'Manager',          'Senior Manager',
 'Représentant Commercial',     'Directeur Technique',
     'Ressources Humaines']
Length: 9, dtype: str
 nombre_experiences_precedentes : [8 1 6 9 0 4 5 2 7 3]
 nombre_heures_travailless : [80]
 annee_experience_totale : [ 8 10  7  6 12  1 17  5  3 31 13  0 26 24 22  9 19  2 23 14 15  4 29 28
 21 25 20 11 16 37

##### <p style="background-color:#b8d1e2;color:black; padding:8px 15px; border-radius:4px; font-size:15px; font-weight:bold;">2.1.1 - Tableaux des variables</p>

| Variable| Descritpion | A garder ? |valeur type |
|:-------|:------|:-------:|-------:  |
|   id_employee    |       |     non  |    numéro   |
|  age   |       |     oui  |       |
|   genre  |       |     oui  |       |
|  revenu_mensuel   |probablement le coût total du salarié|     oui  |       |
|  statut_marital   |       |     oui  |  Célibataire, Marié(e), Divorcé(e)     |
|  departement   |       |     oui  | Commercial, Consulting, Ressources Humaines      |
|  poste   |       |     oui  |       |
|  nombre_experiences_precedentes   |       |     oui  |  nombres     |
|   annees_dans_l_entreprise  |       |     oui  |   années    |
|  annees_dans_le_poste_actuel   |       |     oui  |     années  |
|   nombre_heures_travailless  |       |     non |   toujours 80    |
|   annee_experience_totale  |       |     non  |    années   |


In [86]:
sirh_clean =  sirh.drop(columns=["id_employee","nombre_heures_travailless","annee_experience_totale"])


## <p style="background-color:#1A5276; color:#FFFFFF; padding:15px; border-radius:6px; font-size:22px; font-weight:bold;">3. - DataSet extrait_sondage.csv</p>

### <p style="background-color:#2980B9; color:#FFFFFF; padding:10px 15px; border-radius:5px; font-size:18px; font-weight:bold;">3.1 - Analyse préliminaire</p>

In [68]:
sondage.info()

<class 'pandas.DataFrame'>
RangeIndex: 1470 entries, 0 to 1469
Data columns (total 12 columns):
 #   Column                               Non-Null Count  Dtype
---  ------                               --------------  -----
 0   a_quitte_l_entreprise                1470 non-null   str  
 1   nombre_participation_pee             1470 non-null   int64
 2   nb_formations_suivies                1470 non-null   int64
 3   nombre_employee_sous_responsabilite  1470 non-null   int64
 4   code_sondage                         1470 non-null   int64
 5   distance_domicile_travail            1470 non-null   int64
 6   niveau_education                     1470 non-null   int64
 7   domaine_etude                        1470 non-null   str  
 8   ayant_enfants                        1470 non-null   str  
 9   frequence_deplacement                1470 non-null   str  
 10  annees_depuis_la_derniere_promotion  1470 non-null   int64
 11  annes_sous_responsable_actuel        1470 non-null   int64
dtypes: 

In [69]:
sondage.describe().T

,count,mean,std,min,25%,50%,75%,max
nombre_participation_pee,1470.0,0.793878,0.852077,0.0,0.00,1.0,1.00,3.0
nb_formations_suivies,1470.0,2.799320,1.289271,0.0,2.00,3.0,3.00,6.0
nombre_employee_sous_responsabilite,1470.0,1.000000,0.000000,1.0,1.00,1.0,1.00,1.0
code_sondage,1470.0,1024.865306,602.024335,1.0,491.25,1020.5,1555.75,2068.0
distance_domicile_travail,1470.0,9.192517,8.106864,1.0,2.00,7.0,14.00,29.0
niveau_education,1470.0,2.912925,1.024165,1.0,2.00,3.0,4.00,5.0
annees_depuis_la_derniere_promotion,1470.0,2.187755,3.222430,0.0,0.00,1.0,3.00,15.0
annes_sous_responsable_actuel,1470.0,4.123129,3.568136,0.0,2.00,3.0,7.00,17.0


In [70]:
sondage.head(2)

,a_quitte_l_entreprise,nombre_participation_pee,nb_formations_suivies,nombre_employee_sous_responsabilite,code_sondage,distance_domicile_travail,niveau_education,domaine_etude,ayant_enfants,frequence_deplacement,annees_depuis_la_derniere_promotion,annes_sous_responsable_actuel
0,Oui,0,0,1,1,1,2,Infra & Cloud,Y,Occasionnel,0,5
1,Non,1,3,1,2,8,1,Infra & Cloud,Y,Frequent,1,7


In [71]:
for col in sondage.columns:
    print(f" {col} : {sondage[col].unique()}\n============================")

 a_quitte_l_entreprise : <StringArray>
['Oui', 'Non']
Length: 2, dtype: str
 nombre_participation_pee : [0 1 3 2]
 nb_formations_suivies : [0 3 2 5 1 4 6]
 nombre_employee_sous_responsabilite : [1]
 code_sondage : [   1    2    4 ... 2064 2065 2068]
 distance_domicile_travail : [ 1  8  2  3 24 23 27 16 15 26 19 21  5 11  9  7  6 10  4 25 12 18 29 22
 14 20 28 17 13]
 niveau_education : [2 1 4 3 5]
 domaine_etude : <StringArray>
[          'Infra & Cloud',                   'Autre',
 'Transformation Digitale',               'Marketing',
         'Entrepreunariat',     'Ressources Humaines']
Length: 6, dtype: str
 ayant_enfants : <StringArray>
['Y']
Length: 1, dtype: str
 frequence_deplacement : <StringArray>
['Occasionnel', 'Frequent', 'Aucun']
Length: 3, dtype: str
 annees_depuis_la_derniere_promotion : [ 0  1  3  2  7  4  8  6  5 15  9 13 12 10 11 14]
 annes_sous_responsable_actuel : [ 5  7  0  2  6  8  3 11 17  1  4 12  9 10 15 13 16 14]


##### <p style="background-color:#b8d1e2;color:black; padding:8px 15px; border-radius:4px; font-size:15px; font-weight:bold;">3.1.1 - Tableaux des variables</p>

| Variable| Descritpion | A garder ? |valeur type |
|:-------|:------|:-------:|-------:  |
| a_quitte_l_entreprise|          |oui    | oui/non        |
| nombre_participation_pee |Participation Plan d'Épargne Entreprise|oui    |  0,1,2,3    |
|  nb_formations_suivies      |          |oui    |   0,3,2,5,1,4,6     |
|  nombre_employee_sous_responsabilite      |          |non    |   toujours 1      |
|   code_sondage     |identifiant du sondage |non    |         |
|    distance_domicile_travail    |          |oui    |   km ?      |
|     niveau_education   |          |oui    |  2, 1, 4, 3, 5       |
|     domaine_etude   |          |oui    |         |
|     ayant_enfants   |          |non    |  Tous Y pour yes surement      |
|     frequence_deplacement   |          |oui    | Occasionnel, Frequent, Aucun        |
|     annees_depuis_la_derniere_promotion   |          |oui    |  années       |
|     annes_sous_responsable_actuel   |          |oui    |     années    |


In [72]:
sondage_clean = sondage.drop(columns=["nombre_employee_sous_responsabilite", "code_sondage", "ayant_enfants"])

### <p style="background-color:#2980B9; color:#FFFFFF; padding:10px 15px; border-radius:5px; font-size:18px; font-weight:bold;">3.2 - Analyse a_quitte_l_entreprise</p>

#### <p style="background-color:#AED6F1; color:#1A5276; padding:8px 15px; border-radius:4px; font-size:15px; font-weight:bold;">3.2.1 Graph de répartion</p>

Il y a juste marqué s'il ont quitté l'entreprise, pas si il sont partie d'eux même, ni la nature de la rupture (démission, licensiement, Rupture conventionnelle )

In [73]:
df = sondage["a_quitte_l_entreprise"].value_counts().reset_index()
fig = go.Figure(data=[go.Pie(labels=['Oui','Non'],
                             values=df["count"])])
fig.update_traces(hoverinfo='label+percent', textinfo='value+percent', textfont_size=20,
                  )
fig.show()

#### <p style="background-color:#AED6F1; color:#1A5276; padding:8px 15px; border-radius:4px; font-size:15px; font-weight:bold;">3.2.1 Analyse de ceux qui sont parties</p>

In [74]:
df_demission = sondage[sondage.a_quitte_l_entreprise == "Oui"]
df_demission

,a_quitte_l_entreprise,nombre_participation_pee,nb_formations_suivies,nombre_employee_sous_responsabilite,code_sondage,distance_domicile_travail,niveau_education,domaine_etude,ayant_enfants,frequence_deplacement,annees_depuis_la_derniere_promotion,annes_sous_responsable_actuel
0,Oui,0,0,1,1,1,2,Infra & Cloud,Y,Occasionnel,0,5
2,Oui,0,3,1,4,2,2,Autre,Y,Occasionnel,0,0
14,Oui,0,4,1,19,24,3,Infra & Cloud,Y,Occasionnel,0,3
21,Oui,0,4,1,27,9,4,Infra & Cloud,Y,Occasionnel,0,3
24,Oui,0,2,1,31,6,1,Transformation Digitale,Y,Occasionnel,1,3
...,...,...,...,...,...,...,...,...,...,...,...,...
1438,Oui,1,3,1,2023,9,3,Marketing,Y,Frequent,1,0
1442,Oui,3,3,1,2027,1,4,Transformation Digitale,Y,Occasionnel,2,2
1444,Oui,1,4,1,2032,7,2,Entrepreunariat,Y,Occasionnel,9,8
1452,Oui,2,3,1,2044,1,4,Infra & Cloud,Y,Frequent,0,1


## <p style="background-color:#1A5276; color:#FFFFFF; padding:15px; border-radius:6px; font-size:22px; font-weight:bold;">4. Jointure</p>

In [87]:
df_merge = pd.merge(sirh_clean, eval_clean, left_index=True, right_index=True)
df_merge = pd.merge(df_merge, sondage_clean, left_index=True, right_index=True)
df_merge

,age,genre,revenu_mensuel,statut_marital,departement,poste,nombre_experiences_precedentes,annees_dans_l_entreprise,annees_dans_le_poste_actuel,satisfaction_employee_environnement,...,augementation_salaire_precedente,a_quitte_l_entreprise,nombre_participation_pee,nb_formations_suivies,distance_domicile_travail,niveau_education,domaine_etude,frequence_deplacement,annees_depuis_la_derniere_promotion,annes_sous_responsable_actuel
0,41,F,5993,Célibataire,Commercial,Cadre Commercial,8,6,4,2,...,11 %,Oui,0,0,1,2,Infra & Cloud,Occasionnel,0,5
1,49,M,5130,Marié(e),Consulting,Assistant de Direction,1,10,7,3,...,23 %,Non,1,3,8,1,Infra & Cloud,Frequent,1,7
2,37,M,2090,Célibataire,Consulting,Consultant,6,0,0,4,...,15 %,Oui,0,3,2,2,Autre,Occasionnel,0,0
3,33,F,2909,Marié(e),Consulting,Assistant de Direction,1,8,7,4,...,11 %,Non,0,3,3,4,Infra & Cloud,Frequent,3,0
4,27,M,3468,Marié(e),Consulting,Consultant,9,2,2,1,...,12 %,Non,1,3,2,1,Transformation Digitale,Occasionnel,2,2
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1465,36,M,2571,Marié(e),Consulting,Consultant,4,5,2,3,...,17 %,Non,1,3,23,2,Transformation Digitale,Frequent,0,3
1466,39,M,9991,Marié(e),Consulting,Manager,4,7,7,4,...,15 %,Non,1,5,6,1,Transformation Digitale,Occasionnel,1,7
1467,27,M,6142,Marié(e),Consulting,Tech Lead,1,6,2,2,...,20 %,Non,1,0,4,3,Infra & Cloud,Occasionnel,0,3
1468,49,M,5390,Marié(e),Commercial,Cadre Commercial,2,9,6,4,...,14 %,Non,0,3,2,3,Transformation Digitale,Frequent,0,8


In [88]:
df_merge.info()

<class 'pandas.DataFrame'>
RangeIndex: 1470 entries, 0 to 1469
Data columns (total 25 columns):
 #   Column                                     Non-Null Count  Dtype
---  ------                                     --------------  -----
 0   age                                        1470 non-null   int64
 1   genre                                      1470 non-null   str  
 2   revenu_mensuel                             1470 non-null   int64
 3   statut_marital                             1470 non-null   str  
 4   departement                                1470 non-null   str  
 5   poste                                      1470 non-null   str  
 6   nombre_experiences_precedentes             1470 non-null   int64
 7   annees_dans_l_entreprise                   1470 non-null   int64
 8   annees_dans_le_poste_actuel                1470 non-null   int64
 9   satisfaction_employee_environnement        1470 non-null   int64
 10  niveau_hierarchique_poste                  1470 non-null   

### <p style="background-color:#2980B9; color:#FFFFFF; padding:10px 15px; border-radius:5px; font-size:18px; font-weight:bold;">4.1 - Analyse de la jointure</p>

In [89]:
df_merge.head()

,age,genre,revenu_mensuel,statut_marital,departement,poste,nombre_experiences_precedentes,annees_dans_l_entreprise,annees_dans_le_poste_actuel,satisfaction_employee_environnement,...,augementation_salaire_precedente,a_quitte_l_entreprise,nombre_participation_pee,nb_formations_suivies,distance_domicile_travail,niveau_education,domaine_etude,frequence_deplacement,annees_depuis_la_derniere_promotion,annes_sous_responsable_actuel
0,41,F,5993,Célibataire,Commercial,Cadre Commercial,8,6,4,2,...,11 %,Oui,0,0,1,2,Infra & Cloud,Occasionnel,0,5
1,49,M,5130,Marié(e),Consulting,Assistant de Direction,1,10,7,3,...,23 %,Non,1,3,8,1,Infra & Cloud,Frequent,1,7
2,37,M,2090,Célibataire,Consulting,Consultant,6,0,0,4,...,15 %,Oui,0,3,2,2,Autre,Occasionnel,0,0
3,33,F,2909,Marié(e),Consulting,Assistant de Direction,1,8,7,4,...,11 %,Non,0,3,3,4,Infra & Cloud,Frequent,3,0
4,27,M,3468,Marié(e),Consulting,Consultant,9,2,2,1,...,12 %,Non,1,3,2,1,Transformation Digitale,Occasionnel,2,2


In [101]:
#Calculer la matrice de corrélation
corr_matrix = df_merge.corr(numeric_only=True)



# Créer un masque pour cacher le triangle supérieur (garder diagonale + inférieur)
mask = np.triu(np.ones_like(corr_matrix, dtype=bool), k=1)
corr_masked = corr_matrix.mask(mask)  # remplace par NaN les valeurs masquées

fig = px.imshow(
    corr_masked,
    text_auto=".2f",
    color_continuous_scale="RdBu_r",
    zmin=-1, zmax=1,
    aspect="auto"
)

fig.update_layout(
    width=1000,   # agrandir en largeur
    height=600,  # agrandir en hauteur (carré car matrice carrée)
    xaxis=dict(tickangle=45, dtick=1),  # dtick=1 force l'affichage de CHAQUE label
    yaxis=dict(dtick=1)
)
fig.show()

On vois de trés forte corrélation entre certaine variable, certaine sont attendu.
niveau_hierachique / revenu_mensuel (0.95 - Attendu)
annees_dans_le_poste_actuel / annees_dans_l_entreprise (0.76 - Attendu)
annees_depuis_la_derniere_promotion / annees_dans_l_entreprise (0.62 / moins evident commme corrélation)
annes_sous_responsable_actuel / annees_dans_le_poste_actuel (0.71 / Attendu)


## <p style="background-color:#1A5276; color:#FFFFFF; padding:15px; border-radius:6px; font-size:22px; font-weight:bold;">5. Entrainement</p>
